src/js/main.js

```diff
@@ -31,4 +32,203 @@ Site = {
   },
 };
 
-Site.init();
\ No newline at end of file
+Site.StreamChecker = {
+  statusUrl: 'https://public.radio.co/stations/s0b5e9c02c/status',
+  statusCheckInterval: 5000,
+  streamData: null,
+  init: function() {
+    var _this = this;
+
+    _this.startChecker();
+  },
+
+  startChecker: function() {
+    var _this = this;
+
+
+    _this.checkStreamStatus();
+
+    // Start checker on an interval
+    setInterval(function() {
+      _this.checkStreamStatus();
+    }, _this.statusCheckInterval);
+  },
+
+  // Check the stream status thru ajax and change the player accordingly
+  checkStreamStatus: function() {
+    var _this = this;
+
+    // Make the ajax request
+    $.getJSON(_this.statusUrl, function(data) {
+
+      // Save data in a cache var, just in case
+      _this.streamData = data.status;
+
+      // Check if the stream is offline
+      if(data.status == 'offline') {
+
+        // Trigger the event
+        _this.triggerEvent('streamoffline', data);
+
+      } else if(data.status === 'online')  {  // else  (stream is online)
+
+        // Trigger the event
+        _this.triggerEvent('streamonline', data);
+      }
+    });
+
+  },
+
+  triggerEvent: function(eventName, detail) {
+    var _this =  this;
+
+    console.log('event triggered: ', eventName);
+
+    var customEvent = new CustomEvent(eventName, {
+      detail: detail,
+    });
+
+    document.dispatchEvent(customEvent);
+  },
+
+};
+
+Site.Player = {
+  neverPlayed: true,
+  fadeTime:  2000,
+  streamUrl: 'http://streaming.radio.co/s0b5e9c02c/listen',
+  init: function() {
+    var _this = this;
+
+    // Player DOM elements
+    _this.playerContainer =  document.getElementById('player-container');
```

Comment:
I don't think so, `classList` has good coverage.

It's pretty readable as is, IMO. Specially for someone that doesn't know jQuery, like someone that has only used other languages. jQuery assigns values thru functions, that's not more readable that the universal `=` assignment. 

(You will only keep hating on javascript as long as u keep avoiding it, you need to embrace it. :P )